### DISCUSSION TITLE
Implicit conversion from BigInteger to Long

### DISCUSSION THREAD

**Discussion by andreiyusupau** [mapstruct/mapstruct#3792] [Q&A] [answered]
For some reason implicit conversion from BigInteger to Long uses `BigInteger#longValue`, which could lead to unpredictable results if long is overflowed. Maybe it would be better to add some flag in config, which would enforce usage `BigInteger#longValueExact` which seems to be much safer option?

**Comment by filiphr** [comment, marked as answer]
@andreiyusupau have you looked into the `Mapper#typeConversionPolicy`? This one provides a way to generate a compile warning or error if there is a potential lossy conversion such is converting from `BigInteger` to `long`.

**Reply by andreiyusupau** [reply]
Thanks for your reply. Looks like partial solution. But unfortunately still requires to write conversions manually.

**Reply by filiphr** [reply]
I do understand your point of view. I personally think that it is better to make such a decision during compile time and not have some kind of an exception during runtime. The `Mapper#typeConversionPolicy` offers exactly that, it allows you to explicitly have compile errors and decide on your own how you want to map such potential lossy conversions.
